digits_test.go

```diff
@@ -7,250 +7,334 @@ import (
+		},
+		output: output{
+			sigFigs:    "396006",
+			nonSigFigs: "000",
+			strOut:     "396,006,000.00",
+			head:       "",
+			core:       "396,006",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Millions,
-		v: "77190000.00009",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "487000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "487",
+			nonSigFigs: "000",
+			strOut:     "487,000.00",
+			head:       "",
+			core:       "487",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Millions,
-		v: "-77190000.00009",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "4059000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "4059",
+			nonSigFigs: "000",
+			strOut:     "4,059,000.00",
+			head:       "",
+			core:       "4,059",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "396006000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "45121000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "45121",
+			nonSigFigs: "000",
+			strOut:     "45,121,000.00",
+			head:       "",
+			core:       "45,121",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "487000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "98000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "98",
+			nonSigFigs: "000",
+			strOut:     "98,000.00",
+			head:       "",
+			core:       "98",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "4059000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "-338863000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "338863",
+			nonSigFigs: "000",
+			strOut:     "(338,863,000.00)",
+			head:       "(",
+			core:       "338,863",
+			tail:       ",000.00)",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "45121000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Exact,
+			value:               "1038807",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "1038807",
+			nonSigFigs: "",
+			strOut:     "1,038,807.00",
+			head:       "",
+			core:       "1,038,807.",
```

Comment:
Tis is currently failing.  Shipping a regression issue is not permissible in this project.

```
 go test -timeout 15s github.com/joshuanario/digits
--- FAIL: Test_Core (0.00s)
    digits_test.go:330: 1,038,807 != 1,038,807., 12
--- FAIL: Test_Tail (0.00s)
    digits_test.go:338: .00 != 00, 12
FAIL
FAIL    github.com/joshuanario/digits   0.245s
FAIL
```

Reply:
thanks. This isn't posted for review yet, so I'll comment again when it's actually ready for review.